data/sync.go

```diff
@@ -72,112 +76,123 @@ func SyncFromNode(user User) {
+					})
+				}
 			} else if itemInputLen > 0 { // buy item trade
-				MinLevel := tradeItem.ItemInputs[0].Longs[0].MinValue
-				MaxLevel := tradeItem.ItemInputs[0].Longs[0].MaxValue
-				Name := tradeItem.ItemInputs[0].Strings[0].Value
-				if tradeItem.ExtraInfo == "sword buy request created by loud game" {
+				MinLevel := 0
+				MaxLevel := 0
+				firstItemInput := tradeItem.ItemInputs[0]
+				if len(firstItemInput.Longs) > 0 {
+					MinLevel = firstItemInput.Longs[0].MinValue
+					MaxLevel = firstItemInput.Longs[0].MaxValue
+				}
+				Name := firstItemInput.Strings[0].Value
+				if tradeItem.ExtraInfo == ITEM_BUYREQ_TRDINFO {
 					tItem := ItemSpec{
 						Level: [2]int{MinLevel, MaxLevel},
 						Name:  Name,
 					}
-					nBuySwordTradeRequests = append(nBuySwordTradeRequests, ItemBuyTradeRequest{
-						ID:               tradeItem.ID,
-						TItem:            tItem,
-						Price:            int(pylonOutputAmount),
-						IsMyTradeRequest: isMyTradeRequest,
+					nBuyItemTrdReqs = append(nBuyItemTrdReqs, ItemBuyTrdReq{
+						ID:         tradeItem.ID,
+						TItem:      tItem,
+						Price:      int(pylonOutputAmount),
+						IsMyTrdReq: isMyTrdReq,
 					})
-				} else if tradeItem.ExtraInfo == "character buy request created by loud game" { // character buy request created by loud game
+				} else if tradeItem.ExtraInfo == CHAR_BUYREQ_TRDINFO { // character buy request created by loud game
 					MinXP := 0.0
 					MaxXP := 0.0
-					if len(tradeItem.ItemInputs[0].Doubles) > 0 {
-						MinXP = tradeItem.ItemInputs[0].Doubles[0].MinValue.Float()
-						MaxXP = tradeItem.ItemInputs[0].Doubles[0].MaxValue.Float()
+					if len(firstItemInput.Doubles) > 0 {
+						MinXP = firstItemInput.Doubles[0].MinValue.Float()
+						MaxXP = firstItemInput.Doubles[0].MaxValue.Float()
 					}
 					tCharacter := CharacterSpec{
 						Level: [2]int{MinLevel, MaxLevel},
 						Name:  Name,
 						XP:    [2]float64{MinXP, MaxXP},
 					}
-					nBuyCharacterTradeRequests = append(nBuyCharacterTradeRequests, CharacterBuyTradeRequest{
-						ID:               tradeItem.ID,
-						TCharacter:       tCharacter,
-						Price:            int(pylonOutputAmount),
-						IsMyTradeRequest: isMyTradeRequest,
+					nBuyCharacterTrdReqs = append(nBuyCharacterTrdReqs, CharacterBuyTrdReq{
+						ID:         tradeItem.ID,
+						TCharacter: tCharacter,
+						Price:      int(pylonOutputAmount),
+						IsMyTrdReq: isMyTrdReq,
 					})
 				}
 			} else if itemOutputLen > 0 { // sell item trade
-				inputPylonAmount := tradeItem.CoinInputs[0].Count
-				level, _ := tradeItem.ItemOutputs[0].FindLong("level")
-				name, _ := tradeItem.ItemOutputs[0].FindString("Name")
-				if tradeItem.ExtraInfo == "sword sell request created by loud game" {
+				firstItemOutput := tradeItem.ItemOutputs[0]
+				inputPylonAmount := int64(0)
+				if len(tradeItem.CoinInputs) > 0 {
+					inputPylonAmount = tradeItem.CoinInputs[0].Count
+				}
+				level, _ := firstItemOutput.FindLong("level")
```

Comment:
Shouldn’t we consider multiple output item scenerios too? I can see just handling of 1st item input

Reply:
Currently, loud is only supporting 1 item trading - multiple item trading will make things little complex on UI and it will be done later stage if needed